phoenix-core/src/main/java/org/apache/phoenix/expression/function/JsonExtractPathFunction.java

```diff
@@ -0,0 +1,110 @@
+package org.apache.phoenix.expression.function;
+
+import java.sql.SQLException;
+import java.util.List;
+
+import org.apache.hadoop.hbase.io.ImmutableBytesWritable;
+import org.apache.phoenix.compile.KeyPart;
+import org.apache.phoenix.exception.SQLExceptionCode;
+import org.apache.phoenix.exception.SQLExceptionInfo;
+import org.apache.phoenix.expression.Expression;
+import org.apache.phoenix.parse.FunctionParseNode.Argument;
+import org.apache.phoenix.parse.FunctionParseNode.BuiltInFunction;
+import org.apache.phoenix.schema.IllegalDataException;
+import org.apache.phoenix.schema.json.PhoenixJson;
+import org.apache.phoenix.schema.tuple.Tuple;
+import org.apache.phoenix.schema.types.PDataType;
+import org.apache.phoenix.schema.types.PJsonDataType;
+import org.apache.phoenix.schema.types.PVarcharArray;
+import org.apache.phoenix.schema.types.PhoenixArray;
+import org.apache.phoenix.util.ByteUtil;
+
+@BuiltInFunction(name = JsonExtractPathFunction.NAME, args = {
+        @Argument(allowedTypes = { PJsonDataType.class }),
+        @Argument(allowedTypes = { PVarcharArray.class }) })
+public class JsonExtractPathFunction extends ScalarFunction {
+    public static final String NAME = "JSON_EXTRACT_PATH";
+
+    public JsonExtractPathFunction() {
+        super();
+    }
+
+    public JsonExtractPathFunction(List<Expression> children) {
+        super(children);
+    }
+
+    @Override
+    public boolean evaluate(Tuple tuple, ImmutableBytesWritable ptr) {
+
+        Expression jsonExpression = this.children.get(0);
+        if (!jsonExpression.evaluate(tuple, ptr)) {
+            return false;
+        }
+        if (ptr.getLength() == 0) {
+            return false;
```

Comment:
Return true here. Only return false if the child expression evaluation returns false.